be/src/runtime/primitive_type_infra.h

```diff
@@ -73,6 +73,21 @@ namespace starrocks {
     case type:                    \
         return fun.template operator()<type>(args...);
 
+// Aggregate types
+template <class Functor, class... Args>
+auto type_dispatch_aggregate(PrimitiveType ptype, Functor fun, Args... args) {
+    switch (ptype) {
+        APPLY_FOR_ALL_NUMBER_TYPE(_TYPE_DISPATCH_CASE)
+        _TYPE_DISPATCH_CASE(TYPE_BOOLEAN)
```

Comment:
Could `_TYPE_DISPATCH_CASE` use `std::forward` to avoid copying `args`?
```diff
#define _TYPE_DISPATCH_CASE(type) \
    case type:                    \
-        return fun.template operator()<type>(args...);
+        return fun.template operator()<type>(std::forward(args)...);
```

Reply:
Yes, you could. 

But actually we never pass a rvalue reference into this dispatcher, and almost all parameters are POD type.
BTW, I found it's hard to make compiler happy if I change this piece of code.

Reply:
append && to Args to enable best forwarding
```
auto type_dispatch_aggregate(PrimitiveType ptype, Functor fun, Args... args) {
```

```
auto type_dispatch_aggregate(PrimitiveType ptype, Functor fun, Args&&... args) {
```